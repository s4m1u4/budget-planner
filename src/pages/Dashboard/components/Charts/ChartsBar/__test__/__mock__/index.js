export { TITLE, DATA } from "./mock";
