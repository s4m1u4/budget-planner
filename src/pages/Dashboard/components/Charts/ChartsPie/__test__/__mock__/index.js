export { TITLE, DATA_INCOME } from "./mock";
