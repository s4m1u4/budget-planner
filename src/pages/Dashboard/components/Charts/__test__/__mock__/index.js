export { CATEGORIES, RECORDS } from "./mock";
