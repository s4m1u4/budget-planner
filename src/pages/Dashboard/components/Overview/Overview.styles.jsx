import styled from "@emotion/styled";

export const Amount = styled.div({
  display: "flex",
  justifyContent: "space-between",
});

export const AmountTitle = styled.p({
  fontSize: "18px",
  fontWeight: "500",
});

export const AmountTotal = styled.span({
  fontSize: "18px",
  fontWeight: "500",
});
