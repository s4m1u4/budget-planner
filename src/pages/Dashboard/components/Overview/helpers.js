export const calculateBudgetAmount = (records) => {
  const incomeAmount = records
    .filter((record) => record.type === "income")
    .reduce((acc, value) => acc + +value.amount, 0);

  const expenseAmount = records
    .filter((record) => record.type === "expense")
    .reduce((acc, value) => acc + +value.amount, 0);

  return incomeAmount - expenseAmount;
};
