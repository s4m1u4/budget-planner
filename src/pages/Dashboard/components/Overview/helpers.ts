import { IRecord } from "../../../../types";

export const calculateBudgetAmount: (records: IRecord[]) => number = (
  records: IRecord[]
) => {
  const incomeAmount: number = records
    .filter((record) => record.type === "income")
    .reduce((acc, value) => acc + +value.amount, 0);

  const expenseAmount: number = records
    .filter((record) => record.type === "expense")
    .reduce((acc, value) => acc + +value.amount, 0);

  return incomeAmount - expenseAmount;
};
