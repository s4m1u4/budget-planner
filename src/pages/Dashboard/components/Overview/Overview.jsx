import React from "react";
import currency from "currency.js";
import { Amount, AmountTitle, AmountTotal } from "./Overview.styles";
import { calculateBudgetAmount } from "./helpers";

import {
  Section,
  SectionHeader,
  SectionFooter,
  SectionTitle,
} from "../../Dashboard.styles";

export const Overview = ({ records }) => {
  const budgetAmount = calculateBudgetAmount(records);

  return (
    <Section area="overview">
      <SectionHeader>
        <SectionTitle>Overview</SectionTitle>
      </SectionHeader>
      <SectionFooter>
        <Amount>
          <AmountTitle>Amount:</AmountTitle>
          <AmountTotal>{currency(budgetAmount).format()}</AmountTotal>
        </Amount>
      </SectionFooter>
    </Section>
  );
};
