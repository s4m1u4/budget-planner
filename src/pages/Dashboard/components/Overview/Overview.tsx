import React, { FC } from "react";
import currency from "currency.js";
import { calculateBudgetAmount } from "./helpers";
import { IRecord } from "types";

import {
  Section,
  SectionHeader,
  SectionFooter,
  SectionTitle,
} from "../../Dashboard.styles";
import { Amount, AmountTitle, AmountTotal } from "./Overview.styles";

interface OverviewProps {
  records: IRecord[];
}

export const Overview: FC<OverviewProps> = ({ records }) => {
  const budgetAmount: number = calculateBudgetAmount(records);

  return (
    <Section area="overview">
      <SectionHeader>
        <SectionTitle>Overview</SectionTitle>
      </SectionHeader>
      <SectionFooter>
        <Amount>
          <AmountTitle>Amount:</AmountTitle>
          <AmountTotal>{currency(budgetAmount).format()}</AmountTotal>
        </Amount>
      </SectionFooter>
    </Section>
  );
};
