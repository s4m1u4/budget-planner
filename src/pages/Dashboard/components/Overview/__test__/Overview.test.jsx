import { Overview } from "../Overview";
import { render, screen } from "@testing-library/react";
import { RECORDS } from "./__mock__";

const renderComponent = () => render(<Overview records={RECORDS} />);

describe("Overview component", () => {
  it("snapshot matching", () => {
    const { baseElement } = renderComponent();
    expect(baseElement).toMatchSnapshot();
  });

  describe("should render", () => {
    beforeEach(() => {
      renderComponent();
    });

    it("the title", () => {
      expect(
        screen.getByRole("heading", { name: /overview/i })
      ).toBeInTheDocument();
    });
  });
});
