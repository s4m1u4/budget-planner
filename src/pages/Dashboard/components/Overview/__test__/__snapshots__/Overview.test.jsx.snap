// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview component snapshot matching 1`] = `
<body>
  <div>
    <div
      class="css-1a8z9k1"
    >
      <div
        class="css-13o2qpx"
      >
        <h3
          class="css-1xqtn9i"
        >
          Overview
        </h3>
      </div>
      <div
        class="css-1gk4ta0"
      >
        <div
          class="css-gg4vpm"
        >
          <p
            class="css-15v0w87"
          >
            Amount:
          </p>
          <span
            class="css-15v0w87"
          >
            $5,644.00
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;
